artefacts_utils_rosbag_gpl/ros2bag2video.py

```diff
@@ -0,0 +1,370 @@
+        count = 0
+        serialtype = ''  # Unused
+
+        for line in rosbag2_info:
+            if self.opt_topic in line:
+                parse_line = line.split()
+                for word_index in range(0, len(parse_line)):
+                    if 'Type:' in parse_line[word_index]:
+                        msgtype = parse_line[word_index+1]
+                    if 'Count:' in parse_line[word_index]:
+                        count = int(parse_line[word_index+1])
+                    if 'Serialization' in parse_line[word_index]:
+                        serialtype = parse_line[word_index+2]
+
+        return msgtype, count
+
+    '''
+    The Subscriber Callback.
+    1. Receives images from running ROS2 bag file.
+    2. Get pix_fmt info for ffmpeg process execution.
+    3. Manually converts 16UC1 color encoding to mono16 to avoid cv_bridge
+    conversion error.
+    4. Converts ROS2 image message to OpenCV Mat object.
+    5. Writes individual frame out to file.
+    6. Uses ffmpeg to stitch all frames into a video at user-defined
+    configuration.
+    7. Removes all individual frames.
+    '''
+    def listener_callback(self, msg):
+        self.get_logger().info('Image Received [%i/%i]' %
+                               (self.frame_no, self.count))
+
+        if self.pix_fmt_already_set is not True:
+            self.pix_fmt, self.msg_fmt = self.get_pix_fmt(msg.encoding)
+            self.pix_fmt_already_set = True
+
+        if msg.encoding.find('16UC1') != -1:
+            msg.encoding = 'mono16'
+
+        img = self.bridge.imgmsg_to_cv2(msg, self.msg_fmt)
+
+        filename = str(self.frame_no).zfill(3) + '.png'
+        cv2.imwrite(filename, img)
+
+        '''
+        Once the last frame is reached, combine all individual image frames
+        together to create the video. Remove all individual image frames and
+        kill program once done.
+        Otherwise, continue incrementing frame count.
+        '''
+        if self.frame_no == self.count:
+            p1 = subprocess.Popen([VIDEO_CONVERTER_TO_USE,
+                                  '-framerate',
+                                   str(self.fps),
+                                   '-pattern_type',
+                                   'glob',
+                                   '-i',
+                                   '*.png',
+                                   '-c:v',
+                                   'libx264',
+                                   '-pix_fmt',
+                                   self.pix_fmt,
+                                   self.opt_out_file,
+                                   '-y'])
+            p1.communicate()
+            args = ('rm', '*.png')
+            p2 = subprocess.call('%s %s' % args, shell=True)
+            print('Writing to output file, ' + self.opt_out_file)
+            sys.exit()
+        else:
+            self.frame_no = self.frame_no + 1
+
+def rosbag_to_mp4(rosbag_filepath, topic_name, output_filepath, fps=20):
```

Comment:
This function is the main change (where the function is now what spins everything up when called rather than `__main__`